File: app/test/core/geo/geofence_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:tradeiq_app/core/geo/geofence.dart';

void main() {
  group('isWithinGeofence', () {
    test('returns true when within 50m of the outlet', () {
      const outlet = Coordinates(lat: -26.2041, lng: 28.0473);
      const checkin = Coordinates(lat: -26.20400, lng: 28.0473); // ~11m north
      expect(isWithinGeofence(outlet, checkin), isTrue);
    });

    test('returns false when more than 50m from the outlet', () {
      const outlet = Coordinates(lat: -26.2041, lng: 28.0473);
      const checkin = Coordinates(lat: -26.2100, lng: 28.0473); // ~650m away
      expect(isWithinGeofence(outlet, checkin), isFalse);
    });
  });
}

File: backend/src/lib/geofence.test.ts
import { isWithinGeofence } from './geofence';

describe('isWithinGeofence', () => {
  it('returns true when within 50m of the outlet', () => {
    // ~11m north of the outlet
    const outlet = { lat: -26.2041, lng: 28.0473 };
    const checkin = { lat: -26.20400, lng: 28.0473 };
    expect(isWithinGeofence(outlet, checkin, 50)).toBe(true);
  });

  it('returns false when more than 50m from the outlet', () => {
    const outlet = { lat: -26.2041, lng: 28.0473 };
    const checkin = { lat: -26.2100, lng: 28.0473 }; // ~650m away
    expect(isWithinGeofence(outlet, checkin, 50)).toBe(false);
  });
});
